code/mob/living/critter/sawfly.dm

```diff
@@ -0,0 +1,206 @@
+/*
+->The sawfly, by NightmareChamillian
+This file is the critter itself, and all the custom procs it needs in order to function.
+
+-For the AI, check the critter/AI folder, it should be in sawflyai.dm
+-For the grenade and controller, check code/obj/sawflymisc.dm
+*/
+/mob/living/critter/robotic/sawfly
+
+	name = "Ranodyne antipersonnel microdrone"
+	desc = "A folding antipersonnel drone, made by Ranodyne LLC. It'd be pretty cute if it wasn't trying to kill people."
+	icon = 'icons/obj/ship.dmi'//remnants of it originally being a drone
+	icon_state = "sawfly"
+	flags = TABLEPASS
+
+	var/beeptext = " "
+	var/deathtimer = 0 // for catastrophic failure on death
+	var/isnew = TRUE // for seeing whether or not they will make a new name on redeployment
+	var/sawflynames = list("A", "B", "C", "D", "E", "F", "V", "W", "X", "Y", "Z", "Alpha", "Beta", "Gamma", "Lambda", "Delta")
+	var/beeps = list('sound/machines/sawfly1.ogg','sound/machines/sawfly2.ogg','sound/machines/sawfly3.ogg') // custom noises so they cannot be mistaken for ghostdrones or borgs
+	health = 50 //this value's pretty arbitrary, since it's overridden when they get their healtholders
+	var/fliesnearby = 0 //for rolling chance to beep
```

Comment:
why not make this a local var in the communalbeep proc

Reply:
In the original retaliation code, I made it so the sawflies will play the audio without the communalbeep() proc when attacking, since in combat the chat log tends to go by fast and giving folks an indication of dangerous behavior is pretty important, but forgot to re-add it in the new attackby() proc. I've since re-added it, though.

Reply:
ah! yeah no that can go in communalbeep, apologies.